### PR TITLE
Added all relevant group routes for our MVP

### PR DESCRIPTION
**Pull request by JehronPett**
Added all routes relevant to the member-side interaction between a user and their groups.

### CODE REVIEW

File: app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java [hunk 1]
```diff
@@ -31,9 +33,16 @@
     private static final String API_V2 = "/api/v2";
     private static final String GENERATE_CODE_ROUTE = "/generate/code";
     private static final String MOBILE_AUTH_ROUTE = "/auth/mobile";
+    private static final String JOIN_GROUP_ROUTE = "/join/session";
+    private static final String GET_GROUP_ROUTE = "/sessions/";
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java]:
```suggestion
    private static final String GET_GROUP_ROUTE = "/sessions";
```

There aren't trailing slashes anywhere else, it should be dropped here.

File: app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java [hunk 2]
```diff
@@ -31,9 +33,16 @@
...
     private static final String GET_GROUPS_ROUTE = "/sessions/all";
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java]: alphabetize these?

File: app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java [hunk 3]
```diff
@@ -31,9 +33,16 @@
...
 
-    static UserSession userAuthenticate(GoogleCredentials googleCredentials) throws IOException {
+    private static final String MEMBER_ENDPOINT = "/member";
+    private static final String MEMBERS_ENDPOINT = MEMBER_ENDPOINT + "s";
+    private static final String ADMIN_ENDPOINT = "/admin";
+    private static final String ADMINS_ENDPOINT = ADMIN_ENDPOINT + "s";
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java]: should be alphabetized as well

File: app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java [hunk 4]
```diff
@@ -81,8 +90,63 @@ static String generateCode() throws IOException {
...
+        }
+
+        return null;
+    }
+
+    static List<Group> getAllGroupsAsMember() throws IOException {
+        return getAllGroups(MEMBER_ENDPOINT);
+    }
+
+    static List<Group> getAllGroupsAsAdmin() throws IOException {
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java]: should all of these `static` functions be `private static`? we should have the appropriate identifier for each one

File: app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java [hunk 5]
```diff
@@ -81,8 +90,63 @@ static String generateCode() throws IOException {
...
+        return getAllGroups(ADMIN_ENDPOINT);
+    }
+
+    private static List<Group> getAllGroups(final String roleEndpoint) throws IOException {
+        final String endpoint = R.string.deployed_backed + API_V2 + GET_GROUPS_ROUTE + roleEndpoint;
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java]: `R.string.deployed_backed`? shouldn't it be `deployed_backend`?

File: app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java [hunk 6]
```diff
@@ -101,4 +165,17 @@ static String generateCode() throws IOException {
         return null;
     }
 
+    static boolean leaveGroup(final String id) throws IOException {
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java]: same comment regarding a lack of identifier

File: app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java [hunk 7]
```diff
@@ -101,4 +165,17 @@ static String generateCode() throws IOException {
...
+        final String endpoint = R.string.deployed_backed + API_V2 + GET_GROUP_ROUTE + id + MEMBERS_ENDPOINT;
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java]: personally, i feel like `API_V2` should be part of the constant for the `deployed_backend` string, @AAAstorga what do you guys do on iOS?

File: app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java [hunk 8]
```diff
@@ -158,7 +158,8 @@ protected void onPostExecute(UserSession userSession) {
         protected List<Group> doInBackground(String... strings) {
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java]: can we change this to `runInBackground` lol

File: app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java (hunk 4 above)
Comment (JehronPett) [reply app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java]: No identifier just means package-private, which is suitable for these methods. I can make them public, but that's not necessary.

File: app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java (hunk 5 above)
Comment (JehronPett) [reply app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java]: Who needs Grammarly when you have Young freaking Kim lol

File: app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java (hunk 1 above)
Comment (young-k) [reply app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java]: gotcha, maybe there’s a way to name the variables so this is more clear? not too sure though

File: app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java (hunk 8 above)
Comment (JehronPett) [reply app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java]: It's an overridden method ... I'll mention this to them this summer dw

File: app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java (hunk 7 above)
Comment (JehronPett) [reply app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java]: That makes sense

**Pull request merged**